Won't run on Windows in folder 'drag&drop'

## Describe the bug

Vite won't run on Win10 in the folder with the name `drag&drop` (both dev/build)
Probably since Vite 1.x - https://github.com/vitejs/vite/issues/582

## Logs

![2021-03-17_221633](https://user-images.githubusercontent.com/11320080/111532637-8e63f580-876e-11eb-8277-1d2976a05637.png)

Translation:
"drop\node_modules\.bin\"  - is not internal or external command, executable program, or batch file.

This is an issue in npm, not an issue in vite.
It's reproducible with react-app without vite. Note that you have to uninstall yarn globally for this to work or it would default to building with yarn:
```
>mkdir "dir&withamp"
>cd "dir&withamp"
>npm init react-app my-app
>cd my-app
>npm start
```
Results in:
```shell
> my-app@0.1.0 start
> react-scripts start

"test\my-app\node_modules\.bin\" не является внутренней или внешней
командой, исполняемой программой или пакетным файлом.
node:internal/modules/cjs/loader:926
  throw err;
  ^

Error: Cannot find module 'D:\CODE\react-scripts\bin\react-scripts.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:923:15)
    at Function.Module._load (node:internal/modules/cjs/loader:768:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
npm ERR! code 1
npm ERR! path D:\CODE\apm&test\my-app
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c react-scripts start

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Timson\AppData\Local\npm-cache\_logs\2021-03-27T19_31_11_997Z-debug.log
```

> > This is an issue in npm, not an issue in vite.
> 
> Wait. I have it working with another project using npm - https://preview.adonisjs.com/guides/quick-start
> 
> > npm init adonis-ts-app drag
> 
> then rename the folder to drag&drop
> 
> > npm run dev
> 
> Works!


It seems to work with
`npm run dev`
but gives same error with
`npm run start` :
```
D:\CODE\apm&test\drag&drop>npm run start

> drag@1.0.0 start
> node server.js

node:internal/modules/cjs/loader:926
  throw err;
  ^

Error: Cannot find module 'D:\CODE\apm&test\drag&drop\server.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:923:15)
    at Function.Module._load (node:internal/modules/cjs/loader:768:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
npm ERR! code 1
npm ERR! path D:\CODE\apm&test\drag&drop
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node server.js

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Timson\AppData\Local\npm-cache\_logs\2021-03-27T20_26_18_741Z-debug.log
```

--- a/docs/.vitepress/config.ts
+++ b/docs/.vitepress/config.ts
@@ -212,6 +212,10 @@ export default defineConfig({
               link: '/guide/comparisons'
             },
             {
+              text: 'Troubleshooting',
+              link: '/guide/troubleshooting'
+            },
+            {
               text: 'Migration from v2',
               link: '/guide/migration'
             }

--- a/docs/config/server-options.md
+++ b/docs/config/server-options.md
@@ -145,6 +145,8 @@ Set `server.hmr.overlay` to `false` to disable the server error overlay.
 
 When `server.hmr.server` is defined, Vite will process the HMR connection requests through the provided server. If not in middleware mode, Vite will attempt to process HMR connection requests through the existing server. This can be helpful when using self-signed certificates or when you want to expose Vite over a network on a single port.
 
+Check out [`vite-setup-catalogue`](https://github.com/sapphi-red/vite-setup-catalogue) for some examples.
+
 ::: tip NOTE
 
 With the default configuration, reverse proxies in front of Vite are expected to support proxying WebSocket. If the Vite HMR client fails to connect WebSocket, the client will fallback to connecting the WebSocket directly to the Vite HMR server bypassing the reverse proxies:
@@ -160,8 +162,6 @@ The error that appears in the Browser when the fallback happens can be ignored. 
 - set `server.hmr.port` to a different value from [`server.port`](#server-port)
 
 :::
-
-Checkout [`vite-setup-catalogue`](https://github.com/sapphi-red/vite-setup-catalogue) for some examples.
 
 ## server.watch
 

--- a/docs/config/server-options.md
+++ b/docs/config/server-options.md
@@ -161,6 +161,8 @@ The error that appears in the Browser when the fallback happens can be ignored. 
 
 :::
 
+Checkout [`vite-setup-catalogue`](https://github.com/sapphi-red/vite-setup-catalogue) for some examples.
+
 ## server.watch
 
 - **Type:** `object`

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -1,0 +1,56 @@
+# Troubleshooting
+
+See [rollup's troubleshooting](https://rollupjs.org/guide/en/#troubleshooting) too.
+
+If you didn't make it work, please try posting questions on [GitHub Discussions](https://github.com/vitejs/vite/discussions) or discussioning the issue in #help channel on [Vite Land Discord](https://chat.vitejs.dev).
+
+## CLI
+
+### `Error: Cannot find module 'C:\foo\bar&baz\vite\bin\vite.js'`
+
+Probably the path to your project folder includes `?`.
+There is a bug in `npm` when using Windows and the path including `?` ([npm/cmd-shim#45](https://github.com/npm/cmd-shim/issues/45)).
+
+You will need to either:
+
+- Switch to other package manager (e.g. `pnpm`, `yarn`)
+- Remove `?` from the path to your project
+
+## Dev Server
+
+### Requests are stalled forever
+
+If you are using linux, file descriptor limits and inotify limits may be affecting.
+Because Vite does not bundle most of the files, browsers requests many files and this requires many file descriptors.
+
+To solve this:
+
+- Increase file descriptor limits by `ulimit`
+- Increase the following inotify related limits by `sysctl`
+  - `fs.inotify.max_queued_events`
+  - `fs.inotify.max_user_instances`
+  - `fs.inotify.max_user_watches`
+
+## HMR
+
+### Vite detects a file change but the HMR is not working
+
+Probably you are importing a file with different case.
+For example, `src/foo.js` exists and `src/bar.js` contains:
+
+```js
+import './Foo.js' // should be './foo.js'
+```
+
+Related issue: [#964](https://github.com/vitejs/vite/issues/964)
+
+### Vite does not detect a file change
+
+If you are running Vite with WSL2, Vite cannot watch file changes in some conditions.
+See [`server.watch` option](/config/server-options.md#server-watch).
+
+### A full reload happens instead of HMR
+
+If HMR is not handled by Vite or a plugin, a full reload will happen.
+
+Also if there is a dependency loop, a full reload will happen. To solve this, try removing the loop.

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -34,6 +34,7 @@ To solve this:
   ```
 
 - Increase the following inotify related limits by `sysctl`
+
   ```shell
   # Check current limits
   $ sysctl fs.inotify

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -1,6 +1,6 @@
 # Troubleshooting
 
-See [Rollup's troubleshooting](https://rollupjs.org/guide/en/#troubleshooting) guide for more information too.
+See [Rollup's troubleshooting guide](https://rollupjs.org/guide/en/#troubleshooting) for more information too.
 
 If the suggestions here don't work, please try posting questions on [GitHub Discussions](https://github.com/vitejs/vite/discussions) or in the `#help` channel of [Vite Land Discord](https://chat.vitejs.dev).
 

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -1,15 +1,14 @@
 # Troubleshooting
 
-See [rollup's troubleshooting](https://rollupjs.org/guide/en/#troubleshooting) too.
+See [Rollup's troubleshooting](https://rollupjs.org/guide/en/#troubleshooting) guide for more information too.
 
-If you didn't make it work, please try posting questions on [GitHub Discussions](https://github.com/vitejs/vite/discussions) or discussioning the issue in #help channel on [Vite Land Discord](https://chat.vitejs.dev).
+If the suggestions here don't work, please try posting questions on [GitHub Discussions](https://github.com/vitejs/vite/discussions) or in the `#help` channel of [Vite Land Discord](https://chat.vitejs.dev).
 
 ## CLI
 
 ### `Error: Cannot find module 'C:\foo\bar&baz\vite\bin\vite.js'`
 
-Probably the path to your project folder includes `?`.
-There is a bug in `npm` when using Windows and the path including `?` ([npm/cmd-shim#45](https://github.com/npm/cmd-shim/issues/45)).
+The path to your project folder may include `?`, which doesn't work with `npm` on Windows ([npm/cmd-shim#45](https://github.com/npm/cmd-shim/issues/45)).
 
 You will need to either:
 
@@ -20,8 +19,7 @@ You will need to either:
 
 ### Requests are stalled forever
 
-If you are using linux, file descriptor limits and inotify limits may be affecting.
-Because Vite does not bundle most of the files, browsers requests many files and this requires many file descriptors.
+If you are using Linux, file descriptor limits and inotify limits may be causing the issue. As Vite does not bundle most of the files, browsers may request many files which require many file descriptors, going over the limit.
 
 To solve this:
 
@@ -46,8 +44,7 @@ Related issue: [#964](https://github.com/vitejs/vite/issues/964)
 
 ### Vite does not detect a file change
 
-If you are running Vite with WSL2, Vite cannot watch file changes in some conditions.
-See [`server.watch` option](/config/server-options.md#server-watch).
+If you are running Vite with WSL2, Vite cannot watch file changes in some conditions. See [`server.watch` option](/config/server-options.md#server-watch).
 
 ### A full reload happens instead of HMR
 

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -12,7 +12,7 @@ The path to your project folder may include `?`, which doesn't work with `npm` o
 
 You will need to either:
 
-- Switch to other package manager (e.g. `pnpm`, `yarn`)
+- Switch to another package manager (e.g. `pnpm`, `yarn`)
 - Remove `?` from the path to your project
 
 ## Dev Server
@@ -48,8 +48,7 @@ To solve this:
 
 ### Vite detects a file change but the HMR is not working
 
-Probably you are importing a file with different case.
-For example, `src/foo.js` exists and `src/bar.js` contains:
+You may be importing a file with a different case. For example, `src/foo.js` exists and `src/bar.js` contains:
 
 ```js
 import './Foo.js' // should be './foo.js'

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -23,11 +23,25 @@ If you are using Linux, file descriptor limits and inotify limits may be causing
 
 To solve this:
 
-- Increase file descriptor limits by `ulimit`
+- Increase file descriptor limit by `ulimit`
+
+  ```shell
+  # Check current limit
+  $ ulimit -Sn
+  # Change limit (temporary)
+  $ ulimit -Sn 10000 # You might need to change the hard limit too
+  # Restart your browser
+  ```
+
 - Increase the following inotify related limits by `sysctl`
-  - `fs.inotify.max_queued_events`
-  - `fs.inotify.max_user_instances`
-  - `fs.inotify.max_user_watches`
+  ```shell
+  # Check current limits
+  $ sysctl fs.inotify
+  # Change limits (temporary)
+  $ sudo sysctl fs.inotify.max_queued_events=16384
+  $ sudo sysctl fs.inotify.max_user_instances=8192
+  $ sudo sysctl fs.inotify.max_user_watches=524288
+  ```
 
 ## HMR
 

